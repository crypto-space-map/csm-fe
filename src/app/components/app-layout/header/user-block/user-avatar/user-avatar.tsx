import styled from '@emotion/styled';
import { Avatar, AvatarProps, IconButton, IconButtonProps } from '@mui/material';

type UserAvatarProps = IconButtonProps & AvatarProps;

const StyledAvatar = styled(Avatar)`
  width: 32px;
  height: 32px;
`;

export const UserButtonAvatar = (props: UserAvatarProps) => (
  <IconButton {...props} id="account-block">
    <StyledAvatar {...props} />
  </IconButton>
);
