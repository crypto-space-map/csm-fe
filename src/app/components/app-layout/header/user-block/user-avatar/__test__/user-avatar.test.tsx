import { render } from '@testing-library/react';

import { UserButtonAvatar } from '../user-avatar';

describe('<UserButtonAvatar />', () => {
  it('should match snapshot', () => {
    const { container } = render(<UserButtonAvatar />);
    expect(container).toMatchSnapshot();
  });
  it('should be defined', () => {
    const { container } = render(<UserButtonAvatar />);
    expect(container).toBeDefined();
  });
});
