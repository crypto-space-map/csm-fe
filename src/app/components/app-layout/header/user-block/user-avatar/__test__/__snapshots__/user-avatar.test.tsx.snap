// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserButtonAvatar /> should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    id="account-block"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1223rcr-MuiAvatar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
